### PR TITLE
Windows fixes

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

**Comment by marehr** [comment]
@rrahn simd is also fixed

File: include/seqan/align_parallel/parallel_align_interface.h [hunk 1]
```diff
@@ -110,7 +110,11 @@ struct ParallelAlignmentExecutor
     {
         SEQAN_ASSERT_EQ(length(setH), length(setV));
 
+#if !defined(COMPILER_MSVC)
         using TPos = decltype(length(setH));
+#else
+        using TPos = int;
```
Comment (rrahn) [inline include/seqan/align_parallel/parallel_align_interface.h]: I would rather be more accurate and use `std::make_signed_t<>` on the decltype of `length`

File: include/seqan/align_parallel/parallel_align_interface.h [hunk 2]
```diff
@@ -157,7 +161,11 @@ struct ParallelAlignmentExecutor
...
```
Comment (rrahn) [inline include/seqan/align_parallel/parallel_align_interface.h]: same here.

File: util/cmake/seqan-config.cmake [hunk 3]
```diff
@@ -246,9 +246,8 @@ endif (WIN32)
 # Visual Studio Setup
 if (COMPILER_MSVC OR COMPILER_WINTEL)
   # Enable intrinics (e.g. _interlockedIncrease)
-  # /EHsc will be set automatically for COMPILER_MSVC and COMPILER_WINTEL, but
   # COMPILER_CLANG (clang/c2 3.7) can not handle the /EHsc and /Oi flag
-  set (SEQAN_DEFINITIONS ${SEQAN_DEFINITIONS} /Oi)
+  set (SEQAN_DEFINITIONS ${SEQAN_DEFINITIONS} /EHsc /Oi)
```
Comment (rrahn) [inline util/cmake/seqan-config.cmake:250]: When we add here a required compilation flag we should document this somehow in the manual, for people that build extern with VS, don't we?

**Review by rrahn — CHANGES_REQUESTED** [review]
Some minor remarks left.

**Review by rrahn — APPROVED** [review]

**Pull request merged**